packages/protocol/contracts/identity/Attestations.sol

```diff
@@ -642,16 +642,21 @@ contract Attestations is IAttestations, Ownable, Initializable, UsingRegistry, R
     internal
   {
     IRandom random = IRandom(registry.getAddressForOrDie(RANDOM_REGISTRY_ID));
+    IValidators validators = IValidators(
```

Comment:
Do we need to call into the `Validators` smart contract for this, or can this contract call the precompile directly?

Reply:
We could. I was imagining having Validators expose a wrapper around the precompile to minimize complexity (with assembly in other contracts), though this does increase gas costs.

Reply:
Agree with wanting to cut gas costs, but there is also a small technical problem I just ran into, that the current solution kind of nicely circumvents.

If we use the precompile directly, we can't do ganache tests if we don't modify ethereumjs-vm to handle this precompile (which is not trivial since ganache has no concept of a validator set). By calling into the precompile wrapper in Validators, we can mock Validators and still test this function.

Reply:
As we discussed offline, I think it's probably best to call the precompile directly in a privaet function, and to have a `TestAttestations` contract that inherits from `Attestations` and mocks out the call to the precompile

Reply:
Consider defining a `UsingPrecompiles` contract, e.g. https://github.com/celo-org/celo-monorepo/pull/1292/files#diff-0cffec14a25e6876b387a826c4b4946d

Reply:
It would be nice to add some support for this to ganache. If not, we can't call any part of core contracts that use validators.

Maybe define a list of addresses to serve as validators when launching it